pkg/asset/agent/image/ignitionbase.go

```diff
@@ -0,0 +1,269 @@
+
+	infraEnv := infraEnvAsset.Config
+	clusterImageSet := clusterImageSetAsset.Config
+
+	config := igntypes.Config{
+		Ignition: igntypes.Ignition{
+			Version: igntypes.MaxVersion.String(),
+		},
+		Passwd: igntypes.Passwd{
+			Users: []igntypes.PasswdUser{
+				{
+					Name: "core",
+					SSHAuthorizedKeys: []igntypes.SSHAuthorizedKey{
+						igntypes.SSHAuthorizedKey(infraEnv.Spec.SSHAuthorizedKey),
+					},
+					// PasswordHash: &pwdHash,
+				},
+			},
+		},
+	}
+
+	// a.RendezvousIP = nodeZeroIP
+	// Default to x86_64
+	archName := arch.RpmArch(types.ArchitectureAMD64)
+	if infraEnv.Spec.CpuArchitecture != "" {
+		archName = infraEnv.Spec.CpuArchitecture
+	}
+	a.archName = archName
+
+	releaseImageList, err := releaseImageList(clusterImageSet.Spec.ReleaseImage, archName)
+	if err != nil {
+		return err
+	}
+
+	registriesConfig := &mirror.RegistriesConf{}
+	// registryCABundle := &mirror.CaBundle{}
+	// dependencies.Get(registriesConfig, registryCABundle)
+
+	publicContainerRegistries := getPublicContainerRegistries(registriesConfig)
+
+	logrus.Infof("The release image is %s", clusterImageSet.Spec.ReleaseImage)
+	releaseImageMirror := mirror.GetMirrorFromRelease(clusterImageSet.Spec.ReleaseImage, registriesConfig)
+
+	infraEnvID := uuid.New().String()
+	logrus.Debug("Generated random infra-env id ", infraEnvID)
+	a.infraEnvID = infraEnvID
+
+	osImage, err := getOSImagesInfo(archName)
+	if err != nil {
+		return err
+	}
+	a.osImage = *osImage
+	a.CPUArch = *osImage.CPUArchitecture
+
+	agentTemplateData := &agentTemplateData{
+		PullSecret:                pullSecretAsset.GetPullSecretData(),
+		ReleaseImages:             releaseImageList,
+		ReleaseImage:              clusterImageSet.Spec.ReleaseImage,
+		ReleaseImageMirror:        releaseImageMirror,
+		HaveMirrorConfig:          len(registriesConfig.MirrorConfig) > 0,
+		PublicContainerRegistries: publicContainerRegistries,
+		InfraEnvID:                a.infraEnvID,
+		OSImage:                   a.osImage,
+		Proxy:                     infraEnv.Spec.Proxy,
+	}
+
+	err = bootstrap.AddStorageFiles(&config, "/", "agent/files", agentTemplateData)
+	if err != nil {
+		return err
+	}
+
+	// Set up bootstrap service recording
+	if err := bootstrap.AddStorageFiles(&config,
+		"/usr/local/bin/bootstrap-service-record.sh",
+		"bootstrap/files/usr/local/bin/bootstrap-service-record.sh",
+		nil); err != nil {
+		return err
+	}
+
+	// Use bootstrap script to get container images
+	relImgData := struct{ ReleaseImage string }{
+		ReleaseImage: clusterImageSet.Spec.ReleaseImage,
+	}
+	for _, script := range []string{"release-image.sh", "release-image-download.sh"} {
+		if err := bootstrap.AddStorageFiles(&config,
+			"/usr/local/bin/"+script,
+			"bootstrap/files/usr/local/bin/"+script+".template",
+			relImgData); err != nil {
+			return err
+		}
+	}
+
+	filesToInclude := [...]asset.File{
+		*infraEnvAsset.File,
+		*clusterImageSetAsset.File,
+		*pullSecretAsset.File,
```

Comment:
Do we need the pull secret here only for pulling the release image? If so, since in disconnected env we already provide the entire release payload, can we make it optional?

Reply:
Even assuming the payload is cached already, is it possible to create a cluster without _any_ pull secret?
It seems unlikely that assisted-service would accept this, and even less likely that the installer process it runs to generate the ignition would.

Reply:
So the pull secret is required as part of the config-image, right? Which seems like the proper way to provide the pull secret for creating a cluster. So I wonder why is it needed _also_ in the unconfigured ignition?

Reply:
In every case except a factory-installed image with pre-cached containers (e.g. in the interactive use case) a pull secret is required to pull e.g. the assisted-service container in order to do anything at all. And if one is exists then we'll want to be able use it for the cluster (so the user is not required to provide it twice in the interactive flow).

I don't know how deep the validation we do of the pull secret goes, but I suspect in your case you could easily make up a bogus one that would keep the installer happy, like:

```
{"auths": {"example.com": {"email": "test@example.com", "auth": "this is not a real pull secret"}}}
```